File: php/testcode/benchmark_test_00150.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00150(BenchmarkRequest $req): BenchmarkResponse {
    $path = $req->param('path');
    ob_start();
    readfile($path);
    $content = ob_get_clean();
    return BenchmarkResponse::ok($content);
}

File: rust/testcode/benchmark_test_00150.rs
fn mongo_find(collection: &str, filter: &str) -> String {
    format!("Query on {} with filter: {}", collection, filter)
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let filter = if "release" == "debug" {
        format!("{{\"q\": \"{}\"}}", req.param("q"))
    } else {
        "{\"published\":true}".to_string()
    };
    let result = mongo_find("posts", &filter);
    super::shared::BenchmarkResponse::ok(&result)
}
